wundergraph_cli/src/print_schema/snapshots/wundergraph_cli__print_schema__tests__infer_schema@mysql.snap

```diff
@@ -0,0 +1,150 @@
+---
+source: wundergraph_cli/src/print_schema/mod.rs
+expression: "&s"
+---
+use wundergraph::query_builder::types::{HasMany, HasOne};
+use wundergraph::scalar::WundergraphScalarValue;
+use wundergraph::WundergraphEntity;
+
+table! {
+    comments (id) {
+        id -> Integer,
+        post -> Nullable<Integer>,
+        commenter -> Nullable<Integer>,
+        content -> Text,
+    }
+}
+
+table! {
+    posts (id) {
+        id -> Integer,
+        author -> Nullable<Integer>,
+        title -> Text,
+        datetime -> Timestamp,
+        content -> Nullable<Text>,
+    }
+}
+
+table! {
+    users (id) {
+        id -> Integer,
+        name -> Text,
+    }
+}
+
+allow_tables_to_appear_in_same_query!(
+    comments,
+    posts,
+    users,
+);
+
+
+#[derive(Clone, Debug, Identifiable, WundergraphEntity)]
+#[table_name = "comments"]
+#[primary_key(id)]
+pub struct Comment {
+    id: i32,
+    post: Option<i32>,
+    commenter: Option<i32>,
```

Comment:
That needs to be
```suggestion
    post: Option<HasOne<i32, Post>>,
    commenter: Option<HasOne<i32, User>>,
```

for wundergraph to work. (Same thing for the other relations below)

That looks like foreign key information are not loaded correctly. I would probably start debugging that by looking if those information are correctly loaded [here](https://github.com/weiznich/wundergraph/blob/6c8cb7bd8104c6673ea07ccf3c5feb559979fe41/wundergraph_cli/src/infer_schema_internals/mysql.rs#L37-L80). If not have a look at how diesel_cli handles loading this information. If they are loaded correctly, the filtering [here](https://github.com/weiznich/wundergraph/blob/6c8cb7bd8104c6673ea07ccf3c5feb559979fe41/wundergraph_cli/src/infer_schema_internals/foreign_keys.rs#L6) is probably wrong.